app/src/main/java/com/task/App.kt

```diff
@@ -1,32 +1,18 @@
 package com.task
 
+import android.app.Application
 import android.content.Context
-import androidx.multidex.MultiDexApplication
-import com.task.di.DaggerAppComponent
-import dagger.android.AndroidInjector
-import dagger.android.DispatchingAndroidInjector
-import dagger.android.HasAndroidInjector
-import javax.inject.Inject
+import dagger.hilt.android.HiltAndroidApp
 
 /**
  * Created by AhmedEltaher
  */
-
-open class App : MultiDexApplication(), HasAndroidInjector {
-
-    @Inject
-    lateinit var androidInjector: DispatchingAndroidInjector<Any>
-
-    override fun androidInjector(): AndroidInjector<Any> = androidInjector
+@HiltAndroidApp
+open class App : Application() {
 
     override fun onCreate() {
         super.onCreate()
         context = applicationContext
```

Comment:
I think you do not need to assign applicationContext to context variable here if you need it  activity for example you can make extension function, for Hilt it is all perfect Good Job Man

Reply:
You are right ✔️